internal/cmd/docker/push.go

```diff
@@ -96,11 +98,74 @@ func pushDockerImage(imageName, username, password string, timeout time.Duration
 		return nil
 	}
 
-	// Print the push output
-	_, err = io.Copy(os.Stdout, out)
-	if err != nil {
-		return fmt.Errorf("docker output: %v", err)
+	return logPushProgress(out)
+}
+
+func logPushProgress(reader io.ReadCloser) error {
+	var status string
+	var msg dockermsg.JSONMessage
+	var bar *progressbar.ProgressBar
+
+	decoder := json.NewDecoder(reader)
+	for {
+		// Decode the message from the Docker API output.
+		err := decoder.Decode(&msg)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return fmt.Errorf("failed to decode JSON during Docker image push: %v", err)
+		}
+		if msg.Error != nil {
+			return fmt.Errorf("server error during Docker image push: %s", msg.Error.Message)
+		}
+
+		// Create a new progress bar to display progress whenever the Docker push status changes.
+		if status != msg.Status {
+			status = msg.Status
+			if bar != nil {
+				// Finish the previous progress bar.
+				if err := bar.Finish(); err != nil {
+					return err
+				}
+			}
+			// Set an arbitrary maximum value as the number of steps per status is unknown.
```

Comment:
The fact that the actual progress is unknown and you're just setting an arbitrary value makes me wonder if a progress bar is the right UI to show. It's essentially meaningless in this case. A spinner with the status updating inline might be better. The progress bar docs even describes the case of using a spinner when the progress bar is of unknown length: https://github.com/schollz/progressbar?tab=readme-ov-file#progress-bar-with-unknown-length

Reply:
I actually tried the spinner one before. https://github.com/saucelabs/saucectl/pull/913/commits/baa5d8cbbf25d5ae8ff80c28873234dcfa74892a#diff-1347ecaa302cbb872eb53c259c49c5f51312e6c0ca78a362f07aea15ddf5749eR127
It'd left the last message which is also like meaningless output in the console. 

```
saucectl-local docker push us-west4-docker.pkg.dev/sauce-hto-p-jy6b/sauce-devx-team-sauce/devx-test:2.0
⠋ 2.0: digest: sha256:01f151853941f32bc6547eceb22cd7221a4f1e8dd96d2e9209aad9e6021faff1 size: 1787 (0/-, 0 it/hr) [0s] hr) [0s] 
Successfully pushed the Docker image!
```

Reply:
And `OptionClearOnFinish()` seems can clear anything but it needs `100` as the max which is also an arbitrary value.
The UX is like
```
$ saucectl-local docker push us-west4-docker.pkg.dev/sauce-hto-p-jy6b/sauce-devx-team-sauce/devx-test:2.0
// flashing output and then an empty line
Successfully pushed the Docker image!
```

Reply:
Then what about a different spinner module?

Reply:
One more caveat: docker API output does provide progress. But the current can be larger than total.

```
{"status":"Pushing","progressDetail":{"current":89322496,"total":85157017},"progress":" 89.32MB","id":"dbcbb0f2c34e"}
{"status":"Pushing","progressDetail":{"current":92836352,"total":85157017},"progress":" 92.84MB","id":"dbcbb0f2c34e"}
```

Reply:
Let me do some search for the spinner module. I chose it because we've already used it 😄

Reply:
In the case where current is greater than total, is it like orders of magnitude off? Seems better to use that than an arbitrary number.

Reply:
It's just in "Pushing" status and `total` is in bytes. We still don't know about other statuses. But I've changed spinner type. I think the UX should be better. I'll demo it in stand-up.